common/src/test/java/io/pravega/common/concurrent/ExecutorServiceFactoryTests.java

```diff
@@ -64,7 +64,7 @@ public void testGetDetectionLevel() {
 
     @Test
     public void testScheduledThreadPoolLeak() {
-        testLeaks(factory -> (ThreadPoolScheduledExecutorService) factory.newScheduledThreadPool(1, "test", 1));
+        testLeaks(factory -> (ThreadPoolScheduledExecutorService) factory.newScheduledThreadPool(new ThreadPoolExecutorConfig(1, 1), "test", 1));
```

Comment:
This doesn't seem to be testing the "dynamic change" of threads. I think you may need to add new tests to check if the threads in the pool are changing according to what you expect.